### DISCUSSION TITLE
Label strategy

### DISCUSSION THREAD

**Discussion by tisonkun** [apache/kvrocks#1614] [General]
Currently, we have these sorts of labels:

1. Issue's or PR's area, i.e., which component it is related.

We don't have a top-down component taxonomy, so adding when we meet one:

* [A-build](https://github.com/apache/kvrocks/labels/A-build)
* [A-ci](https://github.com/apache/kvrocks/labels/A-ci)
* [A-cluster](https://github.com/apache/kvrocks/labels/A-cluster)
* [A-docs](https://github.com/apache/kvrocks/labels/A-docs)
* [A-scripting](https://github.com/apache/kvrocks/labels/A-scripting)
* [A-security](https://github.com/apache/kvrocks/labels/A-security)
* [A-site](https://github.com/apache/kvrocks/labels/A-site)
* [A-transaction](https://github.com/apache/kvrocks/labels/A-transaction)

I removed area "server" cause many issues and PRs can be related with "server", especially we don't develop our client. Such a label is too vague to help sorting.

2. Issue's type, i.e., what action is expected to do.

* [bug](https://github.com/apache/kvrocks/labels/bug) to be fixed.
* [crash](https://github.com/apache/kvrocks/labels/crash) to be fixed - this is one kind of bug, but common and critical that deserves special attention.
* [enhancement](https://github.com/apache/kvrocks/labels/enhancement) - generic development task.
* [feature](https://github.com/apache/kvrocks/labels/feature) adds new feature.
* [question](https://github.com/apache/kvrocks/labels/question) - new questions should be opened to Discussions. Keep it for a while for migration.

I removed type "fix" since we should only sorting types on issues, but PRs associated with issues. If we want to sort PRs by their modification, we can use a semantic PR solution (https://github.com/Ezard/semantic-prs).

3. Maintenance labels

* [good first issue](https://github.com/apache/kvrocks/labels/good%20first%20issue)
* [help wanted](https://github.com/apache/kvrocks/labels/help%20wanted)

Other labels require a revisit to work out a clear definition:

* [major decision](https://github.com/apache/kvrocks/labels/major%20decision) maybe we require the author to write a proposal for significant changes instead of a label.
* [need-cherry-pick](https://github.com/apache/kvrocks/labels/need-cherry-pick) how to? A cherry-picking strategy at least sorted by target versions (this is a downside of GitHub issue as a tracking system, though).
* [release notes](https://github.com/apache/kvrocks/labels/release%20notes) this can work. Combined with milestone, the RM know what to include or highlight in the release notes. But do we write release note now?

**Comment by git-hulk** [comment]
> [release notes](https://github.com/apache/kvrocks/labels/release%20notes) this can work. Combined with milestone, the RM know what to include or highlight in the release notes. But do we write release note now?

It looks better to write the release note when tagging the `release notes` label because whoever adds the tag should know precisely why we need to call out the change in the new release.

cc @PragmaTwice @torwig @enjoy-binbin

**Comment by torwig** [comment]
@git-hulk Do we have a dedicated file or something to write release notes to?

**Comment by enjoy-binbin** [comment]
major decision: Usually small PR we can merge directly (one person), if there are some major changes, such as breaking changes, introduction of new features, important bug fixes, etc., we need multiple people to make decisions or reach a consensus

need-cherry-pick: This is a bit redundant, if we use projects to manage it (we add all the PRs that need to be picked to a certain project (when PR was raised and we find out it should be picked, we add it), and review the project to avoid missing things when realeasing). see: https://github.com/orgs/apache/projects/278 and https://github.com/apache/kvrocks/pull/1622

release notes: I saw some unimportant PRs in the report when we last released 2.5.0 (https://github.com/apache/kvrocks/releases/tag/v2.5.0). For some PRs, such as those that need to be known to users, such as major changes, new features, important bug fixes, etc., we can add tags and add explanations when releasing them (for example, if there is a sudden security issue, you can highlight it and then let users to upgrade ASAP)

**Comment by mapleFU** [comment]
Seems we don't have a [A-transaction](https://github.com/apache/kvrocks/labels/A-transaction) part, other look goods to me
